python/cudf/cudf/_libxx/scalar.pyx

```diff
@@ -0,0 +1,220 @@
+import numpy as np
+import pandas as pd
+
+from libc.stdint cimport int8_t, int16_t, int32_t, int64_t
+from libcpp.memory cimport unique_ptr
+from libcpp cimport bool
+from cudf._libxx.lib cimport *
+from cudf._libxx.lib import *
+
+
+cdef class Scalar:
+
+    cdef unique_ptr[scalar] c_value
+
+    def __init__(self, value, dtype=None):
+        """
+        cudf.Scalar: Type representing a scalar value on the device
+
+        Parameters
+        ----------
+        value : scalar
+            An object of scalar type, i.e., one for which
+            `np.isscalar()` returns `True`. Can also be `None`,
+            to represent a "null" scalar. In this case,
+            dtype *must* be provided.
+        dtype : dtype
+            A NumPy dtype.
+        """
+        from cudf.utils.dtypes import to_cudf_compatible_scalar
+
+        value = to_cudf_compatible_scalar(value, dtype=dtype)
```

Comment:
nit: maybe cleaner to return tuple of (value, valid), but not by much.

Reply:
Good point - I think for now it might be a better idea to keep any notion of "validity" within `cudf.Scalar`. We might later consider using a `cudf.NULL` type instead of `None`, in which case `to_cudf_compatible_scalar` can return that instead.